src/main/java/session/FormSession.java

```diff
@@ -98,6 +98,9 @@ public FormSession(String formXml, String restoreXml, String locale, String user
 
         if(instanceContent != null){
             loadInstanceXml(formDef, instanceContent);
+            initialize(false, sessionData);
+        } else {
+            initialize(true, sessionData);
         }
```

Comment:
nit: `initialize(instanceContent == null, sessionData)` so it can be put out of the if